new_guidance_commands/include/new_guidance_commands/NewGuidanceCommands.hpp

```diff
@@ -0,0 +1,129 @@
+#include <mutex>
+// ROS
+#include <ros/ros.h>
+#include <sstream>
+#include <std_srvs/Trigger.h>
+#include <std_msgs/Float32.h>
+// CAV
+#include <cav_msgs/SpeedAccel.h>
+#include <cav_msgs/LateralControl.h>
+#include <cav_msgs/RobotEnabled.h>
+
+namespace new_guidance_commands {
+
+/*!
+ * Main class for the node to handle the ROS interfacing.
+ */
+
+class NewGuidanceCommands {
+    public:
+        /*!
+        * Constructor.
+        * @param nodeHandle the ROS node handle.
+        */
+        NewGuidanceCommands(ros::NodeHandle& nodeHandle);
+
+        /*!
+        * Destructor.
+        */
+        virtual ~NewGuidanceCommands();
+        
+        // Calls speedAccel_Publisher and wrenchEffort_Publisher and lateralControl_Publisher
+        void publisher();
+
+        // runs publish at a desired frequency
+        int rate;
+
+    private:
+        /*!
+        * Reads and verifies the ROS parameters.
+        * @return true if successful.
+        */
+        bool readParameters();
+
+        //! ROS node handle.
+        ros::NodeHandle& nodeHandle_;
+
+        //! ROS speedAccel subscriber.
+        ros::Subscriber speedAccelsubscriber_;
+        std::string speedAccel_subscriberTopic_;
+
+        /*!
+        * ROS topic callback method.
+        * @param message the received message.
+        */
+        void speedAccel_SubscriberCallback(const cav_msgs::SpeedAccel::ConstPtr& msg);
+
+
+
+        //! ROS wrenchEffort subscriber.
+        ros::Subscriber wrenchEffortsubscriber_;
+        std::string wrenchEffort_subscriberTopic_;
+
+        /*!
+        * ROS topic callback method.
+        * @param message the received message.
+        */
+        void wrenchEffort_SubscriberCallback(const std_msgs::Float32::ConstPtr& msg);
+
+
+
+        //! ROS lateralControl subscriber.
+        ros::Subscriber lateralControlsubscriber_;
+        std::string lateralControl_subscriberTopic_;
+
+        /*!
+        * ROS topic callback method.
+        * @param message the received message.
+        */
+        void lateralControl_SubscriberCallback(const cav_msgs::LateralControl::ConstPtr& msg);
```

Comment:
Doesn't need to change, but personally I'm not a fan of mixing `camelCase` and `snake_case` stylistically. But as long as it's consistent across a module I'm not so worried about it.